TO API test TestCDNs fails: Expected only one cdn response

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Traffic Control Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://www.apache.org/security/ regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the #traffic-control channel on the ASF slack (https://s.apache.org/slack-invite)
or Traffic Control mailing lists (https://trafficcontrol.apache.org/mailing_lists).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR.
-->

## I'm submitting a ...
<!-- delete all those that don't apply -->
<!--- security vulnerability (STOP!! - see above)-->
-  bug report

## Traffic Control components affected ...
<!-- delete all those that don't apply -->
-  Traffic Ops

## Current behavior:
<!-- Describe how the bug manifests -->
TestCDNs fails for API v4:

```go
INFO: traffic_ops_test.go:71: 2021-06-17T00:00:33.129707062Z: Using Config values:
                           TO Config File:       ../conf/traffic-ops-test.conf
                           TO Fixtures:          tc-fixtures.json
                           TO URL:               https://localhost:6443
                           TO Session Timeout In Secs:  60
                           DB Server:            localhost
                           DB User:              traffic_ops
                           DB Name:              traffic_ops
                           DB Ssl:               false
               UseIMS:               true
=== RUN   TestCDNs
    cdns_test.go:386: Expected only one cdn response {[] {[]}}
--- FAIL: TestCDNs (1.62s)
FAIL
FAIL    github.com/apache/trafficcontrol/traffic_ops/testing/api/v4     2.130s
FAIL
```

## Expected behavior:
<!-- Describe what the behavior would be without the bug -->
TestCDNs should pass.

## Minimal reproduction of the problem with instructions:
<!--
If the current behavior is a bug, please provide the *STEPS TO REPRODUCE* and
include the applicable TC version.
-->
```shell
go test ./v4 -cfg=../conf/traffic-ops-test.conf -test.v -test.run '^TestCDNs$'
```

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    https://apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/traffic_ops/testing/api/v4/cdns_test.go
+++ b/traffic_ops/testing/api/v4/cdns_test.go
@@ -48,7 +48,7 @@ func TestCDNs(t *testing.T) {
 		header.Set(rfc.IfMatch, etag)
 		UpdateTestCDNsWithHeaders(t, header)
 		GetTestCDNs(t)
-		GetTestCDNsbyDomainName(t)
+		t.Run("get CDNs filtered by Domain Name", GetTestCDNsbyDomainName)
 		GetTestCDNsbyDnssec(t)
 		GetTestCDNsIMSAfterChange(t, header)
 		CreateTestCDNEmptyName(t)
@@ -325,6 +325,7 @@ func UpdateTestCDNs(t *testing.T) {
 	if len(testData.CDNs) < 1 {
 		t.Fatal("Need at least one CDN to test updating CDNs")
 	}
+
 	firstCDN := testData.CDNs[0]
 	opts := client.NewRequestOptions()
 	opts.QueryParameters.Set("name", firstCDN.Name)
@@ -337,12 +338,21 @@ func UpdateTestCDNs(t *testing.T) {
 		t.Fatalf("Expected exactly one CDN to be named '%s', found: %d", firstCDN.Name, len(resp.Response))
 	}
 	remoteCDN := resp.Response[0]
+	originalDomain := remoteCDN.DomainName
 	expectedCDNDomain := "domain2"
 	remoteCDN.DomainName = expectedCDNDomain
 	var alert tc.Alerts
 	alert, _, err = TOSession.UpdateCDN(remoteCDN.ID, remoteCDN, client.RequestOptions{})
 	if err != nil {
 		t.Errorf("cannot update CDN: %v - alerts: %+v", err, alert)
+	} else {
+		defer func(id int, cdn tc.CDN, domain string) {
+			cdn.DomainName = domain
+			alerts, _, err := TOSession.UpdateCDN(id, cdn, client.RequestOptions{})
+			if err != nil {
+				t.Errorf("Unexpected error restoring CDN domain name: %v - alerts: %+v", err, alerts.Alerts)
+			}
+		}(remoteCDN.ID, remoteCDN, originalDomain)
 	}
 
 	// Retrieve the CDN to check CDN name got updated
@@ -359,7 +369,6 @@ func UpdateTestCDNs(t *testing.T) {
 	if respCDN.DomainName != expectedCDNDomain {
 		t.Errorf("results do not match actual: %s, expected: %s", respCDN.DomainName, expectedCDNDomain)
 	}
-
 }
 
 func GetTestCDNs(t *testing.T) {
@@ -383,7 +392,7 @@ func GetTestCDNsbyDomainName(t *testing.T) {
 	opts.QueryParameters.Set("domainName", cdn.DomainName)
 	cdns, reqInf, err := TOSession.GetCDNs(opts)
 	if len(cdns.Response) != 1 {
-		t.Fatalf("Expected only one cdn response %v", cdns)
+		t.Errorf("Expected exactly one CDN to exist with domain name '%s', found: %d", cdn.DomainName, len(cdns.Response))
 	}
 	if err != nil {
 		t.Errorf("cannot get CDN by '%s': %v - alerts: %+v", cdn.DomainName, err, cdns.Alerts)
@@ -448,8 +457,8 @@ func DeleteTestCDNsInvalidId(t *testing.T) {
 	if err == nil {
 		t.Errorf("Expected, no cdn with that key found  but got - alerts: %+v", delResp.Alerts)
 	}
-	if reqInf.StatusCode != http.StatusBadRequest {
-		t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+	if reqInf.StatusCode != http.StatusNotFound {
+		t.Errorf("Expected response status code %d, got %d", http.StatusNotFound, reqInf.StatusCode)
 	}
 }
 

